flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinator.java

```diff
@@ -217,18 +207,11 @@ public void start() throws Exception {
     @Override
     public void close() throws Exception {
         LOG.info("Closing SourceCoordinator for source {}.", operatorName);
-        try {
-            if (started) {
-                context.close();
-                if (enumerator != null) {
-                    enumerator.close();
-                }
+        if (started) {
+            closeQuietly(context);
```

Comment:
Could we use `IOUtils.closeAll` instead? That way we would not loose exception from `context.close()`.

```
IOUtils.closeAll(context, enumerator);
```

Reply:
I get that. Is it mandatory that we do not have any exceptions? Or just that if an exception is thrown in `context.close` we continue with `enumerator.close`. That's what `closeAll` does. The difference is though at the end it will throw the exception from the `context#close` and if both `context.close` and `enumerator.close` throw the second one will be thrown with the first as suppressed.